Framework/DataHandling/src/LoadILLSANS.cpp

```diff
@@ -303,23 +304,25 @@ size_t LoadILLSANS::loadDataIntoWorkspaceFromHorizontalTubes(
 
   Progress progress(this, 0, 1, data.dim0() * data.dim1());
 
-  m_localWorkspace->dataX(firstIndex)
+  m_localWorkspace->mutableX(firstIndex)
       .assign(timeBinning.begin(), timeBinning.end());
 
+  auto sharedXFirstIndex = m_localWorkspace->sharedX(firstIndex);
+
   size_t spec = firstIndex;
   for (size_t i = 0; i < numberOfTubes; ++i) { // iterate tubes
     for (size_t j = 0; j < numberOfPixelsPerTube;
          ++j) { // iterate pixels in the tube 256
       if (spec > firstIndex) {
         // just copy the time binning axis to every spectra
-        m_localWorkspace->dataX(spec) = m_localWorkspace->readX(firstIndex);
+        m_localWorkspace->setSharedX(spec, sharedXFirstIndex);
       }
       // Assign Y
       int *data_p = &data(static_cast<int>(j), static_cast<int>(i), 0);
-      m_localWorkspace->dataY(spec).assign(data_p, data_p + data.dim2());
+      m_localWorkspace->mutableY(spec).assign(data_p, data_p + data.dim2());
 
       // Assign Error
-      MantidVec &E = m_localWorkspace->dataE(spec);
+      auto &E = m_localWorkspace->mutableE(spec);
       std::transform(data_p, data_p + data.dim2(), E.begin(),
                      LoadHelper::calculateStandardError);
```

Comment:
See comment above, maybe the whole block can be written with a single `setHistogram` call.